### PR TITLE
First commit : Base library functions

### PR DESCRIPTION
**Pull request by niio972**
This is the first pull request about this library. It contains minimal functions that permits to work with Handsontable library. 

- Tests need to be implemented.

### CODE REVIEW

File: examples/dataSchema.php [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    [
+                    'id' => null,
+                    'name' => [
+                        'first' => null,
+                        'last' => null],
+                        'address' => null
+                    ]
+                ));
+
+//
```
Comment (Gouga34) [inline examples/dataSchema.php]: remove the "//"

File: src/adapter/HandsontableSimple.php [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+ * HandsontableSimple is a class which extends Handsontable class for native PHP usage
+ * @author Arnaud Charleroy <arnaud.charleroy@inra.fr>
+ * @since 1.0
+ */
+class HandsontableSimple extends \openSILEX\handsontablePHP\classes\Handsontable
+{
+
+    /**
+     * Load all required js librairies
+     * @param array $librairiesPath [ handsontable => ['js => '...', 'css' => ['...']], 'jquery' => ['js => '...', 'css' => ['...']]]
```
Comment (Gouga34) [inline src/adapter/HandsontableSimple.php]: indent the example

File: src/adapter/HandsontableSimple.php [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+     *
+     * @return string string contains html script tag which will be put in head tags
+     */
+    public function loadJSLibraries($jquery = false, $librairiesPath = [])
```
Comment (Gouga34) [inline src/adapter/HandsontableSimple.php]: The @param for the jquery param is missing

File: src/adapter/HandsontableSimple.php [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+        return $js;
+    }
+
+    /**
+     * Load all required css librairies
+     * @param array $librairiesPath [ handsontable => ['js => '...', 'css' => ['...']], 'jquery' => ['js => '...', 'css' => ['...']]]
+     *
+     * @return string contains html link tag which will be put in head tags
+     */
+    public function loadCSSLibraries($jquery = false, $librairiesPath = [])
```
Comment (Gouga34) [inline src/adapter/HandsontableSimple.php]: The @param for the jquery param is missing

File: src/adapter/HandsontableSimple.php [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+                    }
+                }
+            }
+        }
+        return $js;
+    }
+
+    /**
+     * Load all required css librairies
+     * @param array $librairiesPath [ handsontable => ['js => '...', 'css' => ['...']], 'jquery' => ['js => '...', 'css' => ['...']]]
```
Comment (Gouga34) [inline src/adapter/HandsontableSimple.php]: indent the example

File: src/adapter/HandsontableZend.php [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+    protected $view;
+
+
+    public function setView($view)
+    {
+        $this->view = $view;
+    }
+
+    
+    public function loadJSLibraries()
```
Comment (Gouga34) [inline src/adapter/HandsontableZend.php]: this method dont do anything ? You can remove it in this case

File: src/adapter/HandsontableZend.php [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+    {
+//        $this->view->loadViewLibrairies(array('handsontable' => array()));
+    }
+
+    public function render()
+    {
+//        $this->loadJSLibraries();
```
Comment (Gouga34) [inline src/adapter/HandsontableZend.php]: remove the commented line if it is unused

File: src/adapter/HandsontableZend.php [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+        $this->view->jQuery()->addOnLoad($this->generateJavascriptCode());
+    }
+
+    public function loadCSSLibraries()
```
Comment (Gouga34) [inline src/adapter/HandsontableZend.php]: if this method dont do anything, you can remove it or explain why you keep it

File: src/classes/CellConfigDefinition.php [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+    public function isValid()
+    {
+        return (!is_null($this->row) && !is_null($this->column));
+    }
+    /**
+     * @example  {row: 0, col: 0, readOnly: true}
+     */
+    public function jsonSerialize()
+    {
+        return ['row' => $this->row, 'col' => $this->column,'readOnly' => $this->readOnly];
```
Comment (Gouga34) [inline src/classes/CellConfigDefinition.php]: indent to have a better visibility ?

File: src/classes/CellsConfig.php [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+    const CELLS_MODE = 1;
+
+    public function __construct($cellsConfig, $cellMode = CellsConfig::CELLS_MODE)
+    {
+        $this->cellsConfig = $cellsConfig;
+        $this->cellMode = $cellMode;
+    }
+
+    /**
+     * @example
```
Comment (Gouga34) [inline src/classes/CellsConfig.php]: can you explain what is the method doing ? The example is great but i am not sure to really understand the method action only with the example

File: src/classes/Columns.php [hunk 11]
```diff
@@ -0,0 +1,107 @@
...
+     *         return columnMeta;";
+     *    });
+     * @param mixed $columns may be a function or an  array
+     */
+    public function __construct($columns)
+    {
+        $this->columns = $columns;
+    }
+
+    public function jsonSerialize()
```
Comment (Gouga34) [inline src/classes/Columns.php]: add method documentation

File: src/classes/DataSchema.php [hunk 12]
```diff
@@ -0,0 +1,52 @@
...
+        $this->schema = $schema;
+    }
+
+    /**
+     * @example
+     * dataSchema: {id: null, name: {first: null, last: null}, address: null},
+     *
+     * @return string json text
+     */
+    public function jsonSerialize()
```
Comment (Gouga34) [inline src/classes/DataSchema.php]: I don't understand why the method is called jsonSerialize() if it returns the attribute $schema. Shouldn't it be called getSchema ?

File: src/classes/Handsontable.php [hunk 13]
```diff
@@ -0,0 +1,830 @@
...
+     */
+    protected $manualColumnResize = false;
+
+    /**
+     *
+     * @see https://docs.handsontable.com/latest/Options.html#manualRowResize
+     */
+    protected $manualRowResize = false;
+
+    public function __construct($container_name = null, $cellConfig = null)
```
Comment (Gouga34) [inline src/classes/Handsontable.php]: In this constructor, a few things are done. This is not a simple constructor.
You should add the constructor documentation to explain what is done.

File: src/classes/Handsontable.php [hunk 14]
```diff
@@ -0,0 +1,830 @@
...
+    {
+        return $this->rowHeaders;
+    }
+
+    public function getColHeaders()
+    {
+        return $this->colHeaders;
+    }
+
+    public function getContextMenu()
```
Comment (Gouga34) [inline src/classes/Handsontable.php]: in this getter, the method return has differents types. You should documentate the method to explain why

File: src/classes/Handsontable.php [hunk 15]
```diff
@@ -0,0 +1,830 @@
...
+    {
+        $this->height = $height;
+    }
+
+    public function setRowHeaders($rowHeaders)
+    {
+        $this->rowHeaders = $rowHeaders;
+    }
+
+    public function setColHeaders(array $colHeaders)
```
Comment (Gouga34) [inline src/classes/Handsontable.php]: this method is a setter and it returns a value (only in one case). Add documentation to explain why

File: src/classes/Handsontable.php [hunk 16]
```diff
@@ -0,0 +1,830 @@
...
+        if ($this->getLoadAction()) { // generate load custom functions
+            $js_code.= $this->loadTable();
+        }
+        return $js_code;
+    }
+
+    /**
+     * Method which generate the javascript table code needed
+     * @example var hot1 = new Handsontable(container, {
+      data: data(),
```
Comment (Gouga34) [inline src/classes/Handsontable.php]: the "*" is missing (for the others lines too)

File: src/adapter/HandsontableZend.php (hunk 6 above)
Comment (niio972) [reply src/adapter/HandsontableZend.php]: It's a method herited from \openSILEX\handsontablePHP\classes\Handsontable class.

File: src/classes/CellsConfig.php (hunk 10 above)
Comment (niio972) [reply src/classes/CellsConfig.php]: JsonSerialize it's an abstract method provide by \JsonSerializable interface, it's already documented . 
```php
/**
	 * (PHP 5 &gt;= 5.4.0, PHP 7)<br/>
	 * Specify data which should be serialized to JSON
	 * @link http://php.net/manual/en/jsonserializable.jsonserialize.php
	 * @return mixed data which can be serialized by <b>json_encode</b>,
	 * which is a value of any type other than a resource.
	 */
	abstract public function jsonSerialize();
}
```
Should I put a link to this description ? Like @see \JsonSerializable

File: src/adapter/HandsontableSimple.php (hunk 3 above)
Comment (niio972) [reply src/adapter/HandsontableSimple.php]: Done

File: src/classes/Handsontable.php (hunk 13 above)
Comment (niio972) [reply src/classes/Handsontable.php]: Done

File: src/classes/CellsConfig.php [hunk 17]
```diff
@@ -0,0 +1,105 @@
...
+     * Used to save information about cell attribute
+     */
+    const CELL_MODE = 0;
+
+    /**
+     * Used to save information about cells attribute
+     */
+    const CELLS_MODE = 1;
+
+    public function __construct($cellsConfig, $cellMode = CellsConfig::CELLS_MODE) {
```
Comment (Gouga34) [inline src/classes/CellsConfig.php]: add construct definition with @param tags

Comment (niio972) [reply src/classes/CellsConfig.php]: Done

File: src/classes/Handsontable.php [hunk 18]
```diff
@@ -0,0 +1,850 @@
...
+
+    public function getCell() {
+        return $this->cell;
+    }
+
+    public function getCells() {
+        return $this->cells;
+    }
+
+    public function setCells($cells) {
```
Comment (Gouga34) [inline src/classes/Handsontable.php]: In this setter, you change the cells attribute by creating a new CellsConfig. So, what is the expected value of the param $cells ? 
There are no documentation, so I think that the method has a normal behavior and that I have to give a CellsConfig. But you are using the CellsConfig constructor and in the CellsConfig constructor, the expected type of $cell is mixed. In my opinion, you should documentate the setCells, maybe with an example too.

**Pull request closed without merging**